lib/src/view/study/study_list_screen.dart

```diff
@@ -0,0 +1,472 @@
+
+    return PlatformScaffold(
+      appBar: PlatformAppBar(
+        title: title,
+        actions: [
+          AppBarIconButton(
+            icon: const Icon(Icons.tune),
+            semanticsLabel: 'Filter studies',
+            onPressed: () => showAdaptiveBottomSheet<void>(
+              context: context,
+              builder: (_) => _StudyFilterSheet(
+                isLoggedIn: isLoggedIn,
+              ),
+            ),
+          ),
+        ],
+      ),
+      body: _Body(
+        filter: ref.watch(studyFilterProvider),
+      ),
+    );
+  }
+}
+
+class _StudyFilterSheet extends ConsumerWidget {
+  const _StudyFilterSheet({required this.isLoggedIn});
+
+  final bool isLoggedIn;
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final filter = ref.watch(studyFilterProvider);
+
+    return SafeArea(
+      child: Padding(
+        padding: const EdgeInsets.all(16.0),
+        child: Column(
+          mainAxisSize: MainAxisSize.min,
+          children: [
+            const SizedBox(height: 12.0),
+            // If we're not logged in, the only category available is "All"
+            if (isLoggedIn) ...[
+              Filter<StudyCategory>(
+                // TODO l10n
+                filterName: 'Category',
+                filterType: FilterType.singleChoice,
+                choices: StudyCategory.values,
+                choiceSelected: (choice) => filter.category == choice,
+                choiceLabel: (category) =>
+                    Text(studyCategoryL10n(category, context)),
+                onSelected: (value, selected) =>
+                    ref.read(studyFilterProvider.notifier).setCategory(value),
+              ),
+              const PlatformDivider(thickness: 1, indent: 0),
+              const SizedBox(height: 10.0),
+            ],
+            Filter<StudyListOrder>(
+              // TODO l10n
+              filterName: 'Sort by',
+              filterType: FilterType.singleChoice,
+              choices: StudyListOrder.values,
+              choiceSelected: (choice) => filter.order == choice,
+              choiceLabel: (order) => Text(studyListOrderL10n(order, context)),
+              onSelected: (value, selected) =>
+                  ref.read(studyFilterProvider.notifier).setOrder(value),
+            ),
+          ],
+        ),
+      ),
+    );
+  }
+}
+
+class _Body extends StatefulWidget {
+  const _Body({
+    required this.filter,
+  });
+
+  final StudyFilterState filter;
+
+  @override
+  State<_Body> createState() => _BodyState();
+}
+
+class _BodyState extends State<_Body> {
+  String? search;
+
+  final _searchController = SearchController();
+
+  @override
+  void dispose() {
+    super.dispose();
+    _searchController.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return SafeArea(
+      child: Column(
+        children: [
+          Padding(
+            padding: Styles.bodySectionPadding,
+            child: SearchBar(
```

Comment:
We should display a different bar style on iOS.

See how it's done in search scree: https://github.com/lichess-org/mobile/blob/main/lib/src/view/user/search_screen.dart#L108-L115 

Maybe we can write a `PlatformSearchBar` widget?

Reply:
Good idea, done :+1: